### PR TITLE
Add ability to specify HTTP headers in influxdb_listener which will b…

### PR DESCRIPTION
**Pull request by dneray**
…e transformed into tags and added to each metric (#7105)

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/influxdb_listener/influxdb_listener_test.go [hunk 1]
```diff
@@ -155,6 +155,68 @@ func TestWriteBasicAuth(t *testing.T) {
...
+	resp.Body.Close()
+	require.EqualValues(t, 204, resp.StatusCode)
+
+	acc.Wait(1)
+	acc.AssertContainsTaggedFields(t, "cpu_load_short",
+		map[string]interface{}{"value": float64(12)},
+		map[string]string{"host": "server01", "presentMeasurementKey1": "PRESENT_HTTP_VALUE_1", "presentMeasurementKey2": "PRESENT_HTTP_VALUE_2"},
+	)
+
+	// post single message to listener with a database tag in it already. It should be clobbered.
```
Comment (danielnelson) [inline plugins/inputs/influxdb_listener/influxdb_listener_test.go]: Could you split each test case into a separate tests or use a table test?

https://github.com/golang/go/wiki/TableDrivenTests
https://github.com/influxdata/telegraf/blob/7b03a6f4abbf96cbb0c5fc7497071b34c19c9ee8/plugins/inputs/influxdb_listener/influxdb_listener_test.go#L549-L589

Comment (dneray) [reply plugins/inputs/influxdb_listener/influxdb_listener_test.go]: I split the single write and bulk write into separate tests, I think that makes more sense here than using a table test

**Comment by danielnelson** [comment]
I'm having some second thoughts about adding this to the `influxdb_listener` since this plugin is meant to operate as closely as possible to the the InfluxDB API.  What do you think about adding this instead to the `http_listener_v2` input?  When this plugin is setup with `data_format = "influx"` it behaves quite similarly to `influxdb_listener`.

**Comment by dneray** [comment]
I agree, closing this one

**Pull request closed without merging**